src/dms/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlServiceExtensions.cs

```diff
@@ -24,7 +24,26 @@ public static IServiceCollection AddPostgresqlDatastore(
         string connectionString
     )
     {
-        services.AddSingleton((sp) => NpgsqlDataSource.Create(connectionString));
+        services.AddSingleton(sp =>
+        {
+            NpgsqlDataSourceBuilder builder = new(connectionString);
+            var csb = builder.ConnectionStringBuilder;
+
+            // Skip RESET/DISCARD when returning pooled connections, we manage session state explicitly.
+            csb.NoResetOnClose = true;
```

Comment:
No need to spam PostgreSQL with reset and discard, we're not making stateful changes